code/ATMOSPHERICS/components/trinary_devices/filter.dm

```diff
@@ -26,18 +26,18 @@ Filter types:
 
 	ex_node_offset = 5
 
-obj/machinery/atmospherics/trinary/filter/proc/set_frequency(new_frequency)
+/obj/machinery/atmospherics/trinary/filter/proc/set_frequency(new_frequency)
 	radio_controller.remove_object(src, frequency)
 	frequency = new_frequency
 	if(frequency)
 		radio_connection = radio_controller.add_object(src, frequency, RADIO_ATMOSIA)
 
-obj/machinery/atmospherics/trinary/filter/New()
-	if(radio_controller)
+/obj/machinery/atmospherics/trinary/filter/New()
+	if(ticker && ticker.mode == GAME_STATE_PLAYING)
```

Comment:
why not simply move that code from initialize to new()?

Reply:
Because new gets called before radio_controller exists

Reply:
understandable